core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt

```diff
@@ -0,0 +1,1508 @@
+  private var isFirstRun = false
+  protected var actionBar: ActionBar? = null
+  private var tableDrawerAdapter: TableDrawerAdapter? = null
+  private var tableDrawerRight: RecyclerView? = null
+  protected var mainMenu: MainMenu? = null
+  private var tabCallback: ItemTouchHelper.Callback? = null
+  private var bookmarkingDisposable: Disposable? = null
+  private var isBookmarked = false
+  private val toolbarSwipeTouchListener by lazy {
+    object : OnSwipeTouchListener(requireActivity()) {
+      override fun onSwipeBottom() {
+        showTabSwitcher()
+      }
+
+      override fun onSwipeLeft() {
+        if (currentWebViewIndex < webViewList.size - 1) {
+          val current: View = getCurrentWebView()
+          startAnimation(current, R.anim.transition_left)
+          selectTab(currentWebViewIndex + 1)
+        }
+      }
+
+      override fun onSwipeRight() {
+        if (currentWebViewIndex > 0) {
+          val current: View = getCurrentWebView()
+          startAnimation(current, R.anim.transition_right)
+          selectTab(currentWebViewIndex - 1)
+        }
+      }
+
+      override fun onTap(e: MotionEvent?) {
+        val titleTextView = toolbar!!.findFirstTextView() ?: return
+        val hitRect = Rect()
+        titleTextView.getHitRect(hitRect)
+        if (hitRect.contains(e!!.x.toInt(), e.y.toInt())) {
+          mainMenu?.tryExpandSearch(zimReaderContainer.zimFileReader)
+        }
+      }
+    }
+  }
+
+  override fun onActionModeStarted(actionMode: ActionMode, activity: AppCompatActivity):
+    FragmentActivityExtensions.Super {
+    if (this.actionMode == null) {
+      this.actionMode = actionMode
+      val menu = actionMode.menu
+      // Inflate custom menu icon.
+      requireActivity().menuInflater.inflate(R.menu.menu_webview_action, menu)
+      configureWebViewSelectionHandler(menu)
+    }
+    return FragmentActivityExtensions.Super.ShouldCall
+  }
+
+  override fun onActionModeFinished(
+    actionMode: ActionMode,
+    activity: AppCompatActivity
+  ): FragmentActivityExtensions.Super {
+    this.actionMode = null
+    return FragmentActivityExtensions.Super.ShouldCall
+  }
+
+  protected open fun configureWebViewSelectionHandler(menu: Menu?) {
+    menu?.findItem(R.id.menu_speak_text)?.setOnMenuItemClickListener {
+      tts!!.readSelection(getCurrentWebView())
+      actionMode?.finish()
+      true
+    }
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    bottomToolbar = bottom_toolbar
+    noOpenBookButton = go_to_library_button_no_open_book
+    toolBar = toolbar
+    contentFrame = activity_main_content_frame
+    closeAllTabsButton = tab_switcher_close_all_tabs
+    tabSwitcherRoot = activity_main_tab_switcher
+    progressBar = main_fragment_progress_view
+    activityMainRoot = activity_main_root
+    toolbarContainer = fragment_main_app_bar
+    videoView = fullscreen_video_container
```

Comment:
these are all fields, we can't initialise in the field because then we would be trying to find views at class initialisation time, long before any view is inflated.
Ideally we would have no variables and just use the synthetic property